client/src/components/DisplayTable/DisplayTable.tsx

```diff
@@ -164,10 +164,9 @@ export class _DisplayTable extends React.Component<
     if (nextProps !== prevState.initial_props) {
       const new_default_state = get_default_state_from_props(nextProps);
 
-      const prev_sort_still_valid = _.includes(
-        new_default_state.visible_col_keys,
-        prevState.sort_by
-      );
+      const prev_sort_still_valid =
+        _.includes(new_default_state.visible_col_keys, prevState.sort_by) &&
+        prevState.sort_by === new_default_state.visible_col_keys[0];
```

Comment:
It was for grouping by a column that wasn't initially the first column in the table.

For example on Expenditures by Standard Object, if i group by "Standard Object" then group by "all" `sort_by` is still set to "so"

Reply:
i guess I could just not set `unsorted_initial` to false and fix the whole sorting problem that way.

I initially wanted to sort the rows because of this:

> Thing is, if grouping doesn't affect sorting then some of the tables will have weird default orders. In Executive Level the Non-Ex row ends up be between the EX 04 and EX 05 rows. The Voted and Statutory Item rows will be in different orders on different tables. Geographic Region and Transfer Payments by Recipient Region will order the rows differently when grouped by region.

I could undo all my changes to the sorting and open up an issue for after this PR gets merged

Reply:
Hm, right, so making a selection in the group by menu also needs to directly set the new grouping column to be the sorting column. That responsibility needs to belong to the group by feature directly though, for the future maintainability reasons I gave.